wolfcrypt/src/evp.c

```diff
@@ -6878,6 +6878,446 @@ int wolfSSL_EVP_get_hashinfo(const WOLFSSL_EVP_MD* evp,
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[1] = in[j++];
+        inLen--;
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[2] = in[j++];
+        inLen--;
+        if ((res = Base64_SkipNewline(in, &inLen, &j)) != 0) {
+            if (res == BUFFER_E) {
+                break;
+            }
+            else {
+                *outl = 0;
+                return -1;
+            }
+        }
+        e[3] = in[j++];
+        inLen--;
+
+        if (e[0] == '=')
+            pad = 1;
+        if (e[1] == '=')
+            pad = 1;
+        if (e[2] == '=') {
+            pad = 1;
+            pad3 = 1;
+        }
+        if (e[3] == '=') {
+            pad = 1;
+            pad4 = 1;
+        }
+        if (pad3 && !pad4) {
+            *outl = 0;
+            return -1;
+        }
+
+        /* decode four bytes */
+        outsz = sizeof(ctx->data);
+        res = Base64_Decode( e, BASE64_DECODE_BLOCK_SIZE, out, &outsz);
+        if (res == ASN_INPUT_E) {
+            *outl = 0;
+            return -1;
+        }
+
+        *outl += outsz;
+        out   += outsz;
+    }
+    /* copy left data to ctx */
+    if (inLen > 0) {
+        unsigned char el;
+        XMEMSET(ctx->data, 0, sizeof(ctx->data));
+
+        i = 0;
+        while ( inLen > 0) {
+            el = in[j++];
+            if (el== '\n' || el == '\r' || el == ' ') {
+                inLen--;
+                continue;
+            }
+            if (el == '=') {
+                pad = 1;
+            }
+            ctx->data[i++] = el;
+            ctx->remaining++;
+            inLen--;
+        }
+
+        if (pad)
+            return 0;   /* indicates that clients should call DecodeFinal */
+        else
+            return 1;
+
+    }
+    /* if the last data is '\n', remove it */
+    es = in[j - 1];
+    if (es == '\n') {
+        es = (in[j - 2]);
+        if (es == '=')
+            return 0;
+        else
+            return 1;
+    }
+    if (es == '=')
+        return 0;
+    else
+        return 1;
+
+}
+/*  wolfSSL_EVP_DecodeFinal decode remaining data in ctx
+ *  to outputs to out.
+ *  Returns 1 on success, -1 on failure.
+ */
+int  wolfSSL_EVP_DecodeFinal(WOLFSSL_EVP_ENCODE_CTX* ctx,
+                unsigned char*out, int *outl)
+{
+    word32 outsz = 0;
+    word32 inLen;
+    word32 j = 0;
+    int res;
+
+    WOLFSSL_ENTER("wolfSSL_EVP_DecodeFinal");
+
+    if (ctx != NULL) {
+        if (ctx->remaining > 0) {
+            inLen = ctx->remaining;
+            if ((res = Base64_SkipNewline(ctx->data, &inLen, &j)) != 0) {
+                *outl = 0;
+                if (res == BUFFER_E)
+                    return 1;
```

Comment:
Is returning 1 here correct?  Function comment says this should return "-1" on failure.

Reply:
In this case, BUFFER_E means Base64_SkipNewLine could not detect valid data to decode in the buffer.
Since it is not an error, just return as success. Added comments for clarify.